src/changelog.js

```diff
@@ -74,10 +75,38 @@ async function getChangelog({ repository }) {
   }
 }
 
-export async function getChangelogs(pkgs) {
+const CHANGELOG_REGEX = new RegExp(
+  /^(?:(?:update|change|release)(?:s|[ \-_]*(?:logs?|histor(?:y|ies)))|histor(?:y|ies)|release[ \-_]*notes?)(?:\.[\da-z]+)?$/i
+);
+
+function checkChangelogFromFilesList(pkg, filesList) {
+  // Check in fileList
+  if (!filesList || filesList.length <= 0) {
+    return false;
+  }
+
+  const match = filesList.find(file => CHANGELOG_REGEX.test(file.name));
+  if (match) {
+    const url = `${config.jsDelivrCDN}/${pkg.name}@${pkg.version}/${match.name}`;
+    return { changelogFilename: url };
+  }
+
+  return false;
+}
+
+export async function getChangelogs(pkgs, filesLists) {
   const start = Date.now();
 
-  const all = await Promise.all(pkgs.map(getChangelog));
+  const all = await Promise.all(
+    pkgs.map((pkg, index) => {
+      const fromFS = checkChangelogFromFilesList(pkg, filesLists[index]);
+      if (fromFS) {
+        return fromFS;
```

Comment:
It has less impact than I initially anticipated because some packages does not bundle the changelog (most of them do though, because npm auto add this file), and we always rollback to slower method if not found.